### PR TITLE
fix: correctly apply align to `bandPosition`, make center the default align

### PR DESCRIPTION
**Pull request by kanitw**
also don't throw incorrect warning

Fix https://github.com/vega/vega-lite/issues/5505

### CODE REVIEW

File: src/compile/mark/mixins.ts [hunk 1]
```diff
@@ -314,42 +314,39 @@ export function bandPosition(
...
-        return {
-          ...centeredBandPositionMixins,
-          ...nonPosition('size', model, {vgChannel: sizeChannel})
-        };
-      } else if (isValueDef(model.encoding.size)) {
+  const centeredBandPositionMixins = {
+    [vgChannel]: ref.fieldRef(fieldDef, scaleName, {}, {band: 0.5})
+  };
+
+  if (encoding.size != null || markDef.size != null) {
```
Comment (donghaoren) [inline src/compile/mark/mixins.ts]: Should we use `!=` or `!==` here?

File: src/compile/mark/mixins.ts [hunk 2]
```diff
@@ -314,42 +314,39 @@ export function bandPosition(
...
+          [sizeChannel]: {value: markDef.size}
         };
       }
     } else {
-      log.warn(log.message.cannotApplySizeToNonOrientedMark(model.markDef.type));
+      log.warn(log.message.cannotApplySizeToNonOrientedMark(markDef.type));
     }
   }
 
+  if (defaultSizeRef && defaultSizeRef?.value !== undefined) {
```
Comment (donghaoren) [inline src/compile/mark/mixins.ts]: I think `if(defaultSizeRef?.value !== undefined)` is sufficient here

File: src/compile/mark/mixins.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/mark/mixins.ts]: `!=` is useful when you also want to support null and undefined.

File: src/compile/mark/mixins.ts (hunk 2 above)
Comment (kanitw) [reply src/compile/mark/mixins.ts]: Right, I added ? so I can remove the first part, but then forgot to remove lol

File: src/compile/mark/mixins.ts (hunk 1 above)
Comment (donghaoren) [reply src/compile/mark/mixins.ts]: @kanitw said we want to be clearer in terms of semantics with `x !== null && x !== undefined`. Did you guys agree on which one to use?

Comment (domoritz) [reply src/compile/mark/mixins.ts]: `!= null` is a common pattern and I think it's good to use. In all other cases, we want to be explicit.

**Pull request merged**